File: cpp_opencv/threshold/threshold.cpp
#include <opencv2/opencv.hpp>
#include <vector>
#include <stdio.h>

using namespace cv;

std::string photo = "../../photos/dog1.jpg";

int main(){
    Mat image = imread(photo);
    resize(image, image, cv::Size(), 0.2, 0.2);
    imshow("dog", image);

    Mat gray;
    cvtColor(image, gray, COLOR_BGR2GRAY);

    // simple thresholding 
    Mat thresh; 
    threshold(gray, thresh, 100,255, THRESH_BINARY);
    imshow("simple threshold", thresh);

    Mat thresh_inv;
    threshold(gray, thresh_inv, 100, 255, THRESH_BINARY_INV);
    imshow("simple threshold inv", thresh_inv);

    // Adaptive Thresholding
    Mat adap; 
    adaptiveThreshold(gray, adap, 100, ADAPTIVE_THRESH_MEAN_C, THRESH_BINARY, 11, 3);
    imshow("adaptive Thresholding", adap);

    Mat gaussian;
    adaptiveThreshold(gray, gaussian, 100, ADAPTIVE_THRESH_GAUSSIAN_C, THRESH_BINARY, 11, 3);
    imshow("gaussian", gaussian);


    waitKey(0);
    return 0;
}

File: py_opencv/threshold.py
import cv2 as cv
import numpy as np


img = cv.imread("photos/dog1.jpg")
img = cv.resize(img, (int(img.shape[1] * 0.2), int(img.shape[0]*0.2)), interpolation=cv.INTER_LINEAR)
cv.imshow("dog",img)

gray = cv.cvtColor(img, cv.COLOR_BGR2GRAY)
cv.imshow('gray', gray)

# Simple Thresholding 
threshold, thresh = cv.threshold(gray, 100, 255, cv.THRESH_BINARY)
cv.imshow('simple thresholded', thresh)
#   inverse
threshold, thresh_inv = cv.threshold(gray, 100, 255, cv.THRESH_BINARY_INV)
cv.imshow('inverse simple threshold', thresh_inv)

# Adaptive Thresholding 
adaptive_thresh = cv.adaptiveThreshold(gray, 255, cv.ADAPTIVE_THRESH_MEAN_C, cv.THRESH_BINARY, 11, 9)
cv.imshow("adaptive thresholding", adaptive_thresh)
gaussian_thresh = cv.adaptiveThreshold(gray, 255, cv.ADAPTIVE_THRESH_GAUSSIAN_C, cv.THRESH_BINARY, 11, 9)
cv.imshow('gaussian', gaussian_thresh)

cv.waitKey(0)